flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputChannel.java

```diff
@@ -290,17 +288,16 @@ public int unsynchronizedGetNumberOfQueuedBuffers() {
 	 * Parses the buffer as an event and returns the {@link CheckpointBarrier} if the event is indeed a barrier or
 	 * returns null in all other cases.
 	 */
-	@Nullable
-	CheckpointBarrier parseCheckpointBarrierOrNull(Buffer buffer) throws IOException {
+	Optional<CheckpointBarrier> parseCheckpointBarrier(Buffer buffer) throws IOException {
```

Comment:
This is irrelevant change to this commit. Changing `null` to `Optional` hides the real fix/change, so can you please split it into two commits?

Side note, `null` -> `Optional` change can have negative performance impact (unlikely, but possible and previous such changes were going through extra benchmark tests), so it would be good to benchmark it. However we don't have benchmarks with unaligned checkpoints at the moment. Because of that, I would leave it as it is (`@Nullable`) before the feature freeze.

Reply:
Reverted to minimize changes in this PR.